### PR TITLE
Acmorris/terrain exaggeration

### PR DESCRIPTION
**Pull request by acmorris**
@ldanzinger please review. Contains the original PR changes + the updated readme from Michael's comments

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+5. In the `.qml` file, ensure that the main sample object at the root of the QML heirarchy has an ID for reference.
+6. Connect to the QML Slider's `OnValueChanged` signal handler.
+7. Call the `setElevationExaggeration` method of your root object.
+
+## Relevant API
+ - `Surface`
+ - `setElevationExaggeration`
+
+## Tags
+Scenes, 3D, elevation, exaggeration, C++
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/README.md]: C++ tag can be removed

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+2. Create a new function for this called `setExaggerationElevation` that accepts a `double` argument.
+3. Check that m_surface exists using an `if` statement; if this check does not happen, it's possible that the function runs before the scene is even created.
+4. If `m_surface` exists, then access its `setElevationExaggeration` method and pass in the `double` parameter.
+5. In the `.qml` file, ensure that the main sample object at the root of the QML heirarchy has an ID for reference.
+6. Connect to the QML Slider's `OnValueChanged` signal handler.
+7. Call the `setElevationExaggeration` method of your root object.
+
+## Relevant API
+ - `Surface`
+ - `setElevationExaggeration`
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/README.md]: update to Surface::setElevationExaggeration

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  qmlRegisterType<TerrainExaggeration>("Esri.Samples", 1, 0, "TerrainExaggerationSample");
+}
+
+void TerrainExaggeration::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // Create a scene and give it to the SceneView
+  m_sceneView = findChild<SceneQuickView*>("sceneView");
+  Scene* scene = new Scene(Basemap::nationalGeographic(this), this);
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp]: is this the same basemap used in the design? I believe the other samples may use a different basemap. If so, please update and update the screenshots

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        new ArcGISTiledElevationSource(
+          QUrl("http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"),
+          this));
+  // Create a viewpoint where the camera will be
+  Viewpoint initialViewpoint(Point(46.75792111605992, -119.94891542688772, 0, SpatialReference(4326)));
+  // Create the camera object at our initial viewpoint
+  Camera initialViewpointCamera(46.75792111605992, -119.94891542688772, 15000, 60, 40, 0);
+
+  // Set the initial camera/viewpoint combination for the scene
+  m_sceneView->setViewpoint(initialViewpoint);
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp]: this doesn't seem right. You shouldn't need to call `setViewpoint` and then `setViewpointCamera` right after. Can this line be removed?

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+This sample demonstrates how to exaggerate a scene's terrain surface.
+
+![](screenshot.png)
+
+## How to use the sample
+Click and drag the slider at the top to change how exaggerated the terrain becomes.
+
+## How it works
+1. Create a scene.
+2. Add a surface feature layer.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/README.md]: the surface isn't a feature layer. You could say something like "Create and add a surface with an ArcGISTiledElevationService"

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+3. Surfaces have a property known as `elevationExaggeration` which allows you to change the surface exaggeration. Set this value as an attribute of the `Surface` object.
+4. In this case, we have a slider object with values between 1 and 10. We've bound the surface's elevationExaggeration to the slider's value, which allows for a dynamic demonstration of the exaggeration effects.
+
+## Relevant API
+ - `Surface`
+ - `elevationExaggeration`
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/README.md]: change to Surface.elevationExaggeration

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp (hunk 3 above)
Comment (acmorris) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/TerrainExaggeration/TerrainExaggeration.cpp]: Java (which is what I was using as a reference) uses natGeo basemap, iOS/macOS use streets. Which one do you prefer?

**Comment by ldanzinger** [comment]
@acmorris doesn't seem like the basemap discussion got any traction. Can you please update the PR so they both use the same one (C++ and QML). Either make both Topo or both NatGeo

**Comment by acmorris** [comment]
@ldanzinger basemaps are now matching

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/TerrainExaggeration.qml [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+    property real scaleFactor: System.displayScaleFactor
+
+    SceneView {
+        id: sceneView
+        anchors.fill: parent
+
+        // add a Scene to the SceneView
+        Scene {
+            // add the BasemapTopographic basemap to the scene
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/TerrainExaggeration.qml]: comment is out of date

Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_QMLSamples/Scenes/TerrainExaggeration/TerrainExaggeration.qml]: also, please update the screenshot

**Pull request merged**